### PR TITLE
Removing dependence on git in gemspec

### LINKED ISSUE #1214: Remove git dependence in gemspec
Currently the gemspec has a dependency on git by using the `git ls-files` syntax.

We could switch it to use the `Dir` syntax which removes the dependence on git

### PR DESCRIPTION
**Pull request by ochagata**
This is a PR to fix #1214

### CODE REVIEW

**Comment by dblock** [comment]
I'll merge it. Can you squash these commits please? Thx.

File: grape.gemspec [hunk 1]
```diff
@@ -36,8 +36,8 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'benchmark-ips'
   s.add_development_dependency 'rubocop', '0.35.1'
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
+  s.files         = Dir['**/*'].keep_if { |file| File.file?(file) }
+  s.test_files    = Dir['{test,spec,features}/**/*']
+  s.executables   = Dir['bin/*'].map { |f| File.basename(f) }
```
Comment (dblock) [inline grape.gemspec]: We actually don't have any executables, lets just remove this?

File: grape.gemspec [hunk 2]
```diff
@@ -36,8 +36,8 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'benchmark-ips'
   s.add_development_dependency 'rubocop', '0.35.1'
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
+  s.files         = Dir['**/*'].keep_if { |file| File.file?(file) }
+  s.test_files    = Dir['{test,spec,features}/**/*']
```
Comment (dblock) [inline grape.gemspec]: We don't have all these paths, lets fix it too?

Comment (ochagata) [inline grape.gemspec]: Agreed

**Comment by ochagata** [comment]
Squashed and fixed based on your comments, @dblock

**Comment by dblock** [comment]
Merged, thanks!

**Pull request merged**